### DISCUSSION TITLE
Basic Method invocation Question

### DISCUSSION THREAD

**Discussion by AlexGoico** [hylang/hy#2619] [Q&A] [answered]
In Clojure, method invocation can be as follows: `(. instance-expr method-symbol args*)`

In Hy, I see from docs I can invoke as:
```
(.method-symbol instance-expr args*)
```
or
```
(instance-expr.method-symbol args*)
```

but for whatever reason not:
```
(. instance-expr method-symbol args*)
```
I must instead return the method reference then add `args*`
```
((. instance-expr method-symbol) args*)
```
Am I missing something or is this something intentional of the language design? I don't see much clarity on https://hylang.org/hy/doc/v1.0.0/tutorial#functions-classes-and-modules.

Tested this using loguru as follows in hy 1.0.0 interpreter:
```
=> (import loguru [logger])
=> (. logger info "hi")
Traceback (most recent call last):
  File "stdin-931ab003c23f8ae59707682229501bee380215b8", line 1
    (. logger info "hi")
                   ^
hy.errors.HySyntaxError: parse error for pattern macro '.': got unexpected token: hy.models.String('hi'), expected: end of macro call
```

**Comment by Kodiologist** [comment, marked as answer]
> is this something intentional of the language design?

It is, yes. The reason that `(. a b c)` doesn't compile to `b.a(c)` is that it compiles to something else, namely `a.b.c`. See http://hylang.org/hy/doc/v1.0.0/api#%2E.

**Reply by AlexGoico** [reply]
Appreciate the answer to such a basic question. Thanks.
